server/adaptors/dslPlugin.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import {
+  DSL_ENDPOINT
+} from '../../common/constants/shared';
+
+export const DSLPlugin = function(Client, config, components) {
```

Comment:
> does autocomplete cache request results? for example if user types `source=indexA | where` then modifies `indexA` to `indexB`, would the suggested fields refresh to fields of `indexB`?

so far, i haven't implemented something like that since i'm parsing the string based on the last word in the query.

Reply:
> i don't think you'd need a DSLPlugin? this is used to call backend APIs, but we don't have `_plugin/_dsl`. DSL call functionality is built in with the opensearch client

that's right, i'll remove those plugins.